### PR TITLE
[lua] Add regression test for pcall return wrapping

### PR DESCRIPTION
**Pull request by jdonaldson**
## Summary

- Follow-up to #12601
- Adds a regression test (`Issue7350`) for the `in_pcall` guard that ensures `do return end` wrapping is preserved inside try-catch blocks (pcall callbacks), where the compiler appends `return _hx_pcall_default` after user code.
- Also tests that tail returns use bare `return` without wrapping.

## Test plan

- [x] `make haxe` builds successfully
- [x] All 11,711 Lua unit tests pass

### CODE REVIEW

File: tests/unit/src/unit/issues/Issue7350.hx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+	function returnInTryCatch(x:Int):Int {
+		try {
+			if (x > 0) {
+				return x * 2;
+			}
+		} catch (e:Dynamic) {
+			return -1;
+		}
+		return 0;
+	}
```
Comment (tobil4sk) [inline tests/unit/src/unit/issues/Issue7350.hx:21]: For this function, haxe currently generates:
```lua
___Main_Main_Fields_.returnInTryCatch = function(x) 
  local _hx_status, _hx_result = pcall(function() 
  
      if (x > 0) then 
        do return x * 2 end;
      end;
    return _hx_pcall_default
  end)
  if not _hx_status then 
    local _g = _hx_result;
    return -1;
  elseif _hx_result ~= _hx_pcall_default then
    return _hx_result
  end;
  return 0
end
```

from what I can tell in this case the `do` `end` doesn't make much of a difference?

File: tests/unit/src/unit/issues/Issue7350.hx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+		var result = x + 1;
+		return result;
+	}
+
+	function test() {
+		// Return inside try-catch (pcall) must use do...end wrapper
+		eq(10, returnInTryCatch(5));
+		eq(0, returnInTryCatch(-1));
+
+		// Tail return should not use do...end wrapper
```
Comment (tobil4sk) [inline tests/unit/src/unit/issues/Issue7350.hx]: This test doesn't really do much, because even if there is a `do ... end` it's still going to pass. It's most useful to test if there is a case where the compiler missing out `do ... end` would result in broken output.

Comment (jdonaldson) [reply tests/unit/src/unit/issues/Issue7350.hx]: Good point — rewrote the test to exercise actual pcall return propagation edge cases instead. Now covers returning null/false/zero through pcall (must not be confused with `_hx_pcall_default` sentinel), fall-through where try body has no return (code after try-catch must execute), nested try-catch, try-catch in loops, and exception handling paths.
